actors/miner/src/lib.rs

```diff
@@ -2995,6 +2996,301 @@ impl Actor {
+                        let sector_infos = sectors
+                            .load_for_proof(
+                                &BitField::union(&all_sectors),
+                                &BitField::union(&all_ignored),
+                            )
+                            .map_err(|e| {
+                                e.downcast_default(
+                                    ExitCode::USR_ILLEGAL_STATE,
+                                    "failed to load sectors for post verification",
+                                )
+                            })?;
+
+                        // Find the proving period start for the deadline in question.
+                        let mut pp_start = current_deadline.period_start;
+                        if current_deadline.index < params.from_deadline {
+                            pp_start -= policy.wpost_proving_period
+                        }
+                        let target_deadline =
+                            new_deadline_info(policy, pp_start, params.from_deadline, rt.curr_epoch());
+
+                        if !verify_windowed_post(
+                            rt,
+                            target_deadline.challenge,
+                            &sector_infos,
+                            window_proof.proofs.clone(),
+                        )
+                        .map_err(|e| e.wrap("window post failed"))?
+                        {
+                            return Err(actor_error!(
+                                illegal_argument,
+                                "invalid post was submitted"
+                            )
+                            .into());
+                        }
+                        Ok(())
+                    })
+                    .map_err(|e| {
+                        e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "while removing partitions")
+                    })?;
+            }
+
+            if !deadline_available_for_compaction(
+                policy,
+                current_deadline.period_start,
+                params.to_deadline,
+                rt.curr_epoch(),
+            ) {
+                return Err(actor_error!(
+                    forbidden,
+                    "cannot move to deadline {} during its challenge window, \
+                        or the prior challenge window,
+                        or before {} epochs have passed since its last challenge window ended",
+                    params.to_deadline,
+                    policy.wpost_dispute_window
+                ));
+            }
+
+            if !deadline_available_for_move(
+                policy,
+                params.from_deadline,
+                params.to_deadline,
+                current_deadline.index,
+            ) {
+                return Err(actor_error!(
+                    forbidden,
+                    "cannot move to a deadline which is further from current deadline"
+                ));
+            }
+
```

Comment:
At this point, now all the policy logic is out of the way, please move the subsequent state manipulation into a method or function in deadline_state.rs, where it can also be tested directly.

Reply:
Done.